codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/CodegenVisitor.kt

```diff
@@ -50,20 +50,28 @@ class CodegenVisitor(context: PluginContext) : ShapeVisitor.Default<Unit>() {
         LOGGER.info("Discovering KotlinIntegration providers...")
         integrations = ServiceLoader.load(KotlinIntegration::class.java, classLoader)
             .onEach { integration -> LOGGER.info("Loaded KotlinIntegration: ${integration.javaClass.name}") }
-            .filter { integration -> integration.enabledForService(context.model, settings) } // TODO: Change so we don't filter until previous integrations model modifications are complete
-            .onEach { integration -> LOGGER.info("Enabled KotlinIntegration: ${integration.javaClass.name}") }
             .sortedBy(KotlinIntegration::order)
+            .toMutableList()
```

Comment:
design suggestion: instead of making this a `MutableList`, it might be better to have `configuredIntegrations` and `enabledIntegrations` (`configuredIntegrations - disabledIntegrations`)

Reply:
example:

```
        configuredIntegrations = ServiceLoader.load(KotlinIntegration::class.java, classLoader)
...
        val enabledIntegrations = configuredIntegrations.mapNotNull { integration ->
            if (integration.enabledForService(resolvedModel, settings)) {
                println("Executing KotlinIntegration: ${integration.javaClass.name}")
                resolvedModel = integration.preprocessModel(resolvedModel, settings)
                integration
            } else {
                null
            }
        }
```